mod-di-converter-storage-server/src/main/java/org/folio/services/migration/ProfileMigrationServiceImpl.java

```diff
@@ -37,17 +41,37 @@ public class ProfileMigrationServiceImpl implements ProfileMigrationService {
   public Future<Boolean> migrateDataImportProfiles(Map<String, String> headers, Context context) {
     String tenantId = new OkapiConnectionParams(headers).getTenantId();
     LOGGER.info("Profile migration started...");
-    return profileWrapperDao.checkIfDataInTableExists(tenantId)
-      .compose(isDataPresent -> {
-        if (!isDataPresent) {
-          return runScript(tenantId, INIT_WRAPPERS)
-            .compose(ar -> runScript(tenantId, UPDATE_SCHEMA_FOR_MIGRATION));
+
+    return profileWrapperDao.getLinesCount(tenantId, SYSTEM_TABLE_NAME)
+      .compose(isRowCount -> {
+        if (isRowCount == 0) {
+          return profileWrapperDao.checkIfDataInTableExists(tenantId)
+            .compose(isDataPresent -> processMigration(isDataPresent, tenantId));
         } else {
-          LOGGER.info("migrateDataImportProfiles:: Migration will not execute. profile_wrappers table is NOT empty already.");
+          LOGGER.info("migrateDataImportProfiles:: Migration already executed.");
           return Future.succeededFuture(true);
         }
       })
-      .onFailure(th -> LOGGER.error("migrateDataImportProfiles:: Something happened during the profile migration", th));
+      .onFailure(th -> {
+        LOGGER.error("migrateDataImportProfiles:: Something happened during the profile migration", th);
+        throw new CompletionException(th);
```

Comment:
why throw this? I don't think `onFailure` needs a return object.

Reply:
Fixed, thanks